packages/web-app-files/src/components/SideBar/Shares/FileLinks.vue

```diff
@@ -450,40 +456,44 @@ export default defineComponent({
       if (this.hasShareJail && path === '/') {
         path = `/${this.resource.name}`
       }
-      await this.addLink({
-        path,
-        client: this.$client,
-        storageId: this.resource.fileId || this.resource.id,
-        params
-      }).catch((e) => {
+      try {
+        await this.addLink({
+          path,
+          client: this.$client,
+          storageId: this.resource.fileId || this.resource.id,
+          params
+        })
+      } catch (e) {
         onError(e)
         console.error(e)
         this.showMessage({
           title: this.$gettext('Failed to create link'),
           status: 'danger'
         })
-      })
+        return
```

Comment:
prevents a fall through to the success message when the request resulted in an error status code. previously an error would cause both the error and the success message. :trollface: